src/test/java/IvanFFavoriteTests.java

```diff
@@ -0,0 +1,109 @@
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        homePageIvan.favorite();
+        homePageIvan.playSongFromfav();
+        homePageIvan.currentQueue();
+    }
+    @Test
+    public void deleteSongFromFav() throws InterruptedException{
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        homePageIvan.favorite();
+        homePageIvan.deleteSongFromfav();
+
+    }
+    @Test
+    public void addSongToFavHeart() throws InterruptedException{
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        AllSongsIvan allSongsIvan = new AllSongsIvan(getDriver());
+        int initialNumberOfFav = homePageIvan.getNumberfavourite();
+        homePageIvan.clickOnAllSongs();
+        allSongsIvan.addToFavoriteHeart();
+        int updatedNumberOfFav = homePageIvan.getNumberfavourite();
+        System.out.println("Number of favorites before: "+initialNumberOfFav+" New Number Of Favorites:  "+updatedNumberOfFav);
+
+    }
+    @Test
+    public void downloadAllSongFromFav() throws InterruptedException {
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        AllSongsIvan allSongsIvan = new AllSongsIvan(getDriver());
+        homePageIvan.clickOnAllSongs();
+        allSongsIvan.addToFavoriteHeart();
+        homePageIvan.favorite();
+        homePageIvan.downloadAllSongFromFav();
+
+    }
+    @Test
+    public void downloadOneSongFromFav() throws InterruptedException {
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        AllSongsIvan allSongsIvan = new AllSongsIvan(getDriver());
+        homePageIvan.clickOnAllSongs();
+        allSongsIvan.addToFavoriteHeart();
+        homePageIvan.favorite();
+        Thread.sleep(2000);
+        homePageIvan.contextDownload();
+        Thread.sleep(2000);
+        Assert.assertTrue(homePageIvan.isSongDownloaded("HoliznaCC0 - Waiting On A Train.mp3"));
+    }
+    @Test
+    public void checkIfFavEmpty() throws InterruptedException {
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        homePageIvan.favorite();
+        Thread.sleep(2000);
+        homePageIvan.emptyFavPlay();
+
+    }
+//   Drag and Droop method not working with the Koel App
+//    @Test
+//    public void dragAndDropSongs() throws InterruptedException {
+//        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+//        HomePageIvan homePageIvan = loginPage.login();
+//        AllSongsIvan allSongsIvan = new AllSongsIvan(getDriver());
+//        homePageIvan.clickOnAllSongs();
+//        Thread.sleep(2000);
+//        allSongsIvan.dragAndDropFav();
+//        Thread.sleep(2000);
+//
```

Comment:
**I recommend not using code comments:** for, example, when you have a code that cannot be understood, then comments are not the solution. Try to rewrite the code or rename fields and other elements in a way that the reader will understand the action that you are doing there.
![image](https://user-images.githubusercontent.com/104761482/206839550-5fa01ad1-fd3c-444a-876a-fc08eac56427.png)